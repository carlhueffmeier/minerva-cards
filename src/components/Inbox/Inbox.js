import React, { Component } from 'react';
import Button from 'components/Button';
import NewDeck from 'components/NewDeck';
import { Link } from 'react-router-dom';

export default class Inbox extends Component {
  goToReview() {
    this.props.history.push(`/review`);
  }

  render() {
    return (
      <div>
        <h1>Inbox</h1>
        <ul className="deck-list">
          {this.props.decks.map(deck => (
            <li key={deck.id}>
              <Link to={`/decks/${deck.id}`}>{deck.title}</Link>
            </li>
          ))}
        </ul>

        <NewDeck />

        <Button onClick={this.goToReview.bind(this)}>Start Learning</Button>
      </div>
    );
  }
}
