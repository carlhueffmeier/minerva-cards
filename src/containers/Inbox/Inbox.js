import React from 'react';
import { Button } from 'components';
import { connect } from 'react-redux';
import { selectors as deckSelectors } from 'state/modules/decks';
import { NewDeck } from 'components';
import { Link } from 'react-router-dom';

function goToReview() {
  console.log('changing route');
}

export function Inbox(props) {
  return (
    <div className="App">
      <h1>Inbox</h1>
      <p>Filter by: </p>
      <ul>
        <li>
          <a href="#">Due Decks</a>
        </li>
        <li>
          <a href="#">Active Decks</a>
        </li>
        <li>
          <a href="#">Inactive Decks</a>
        </li>
      </ul>
      <ul className="deck-list">
        {props.decks.map(deck => (
          <li key={deck.id}>
            <Link to={`/decks/${deck.id}`}>{deck.title}</Link>
          </li>
        ))}
      </ul>

      <NewDeck />

      <Button onClick={goToReview}>Start Learning</Button>
    </div>
  );
}
function mapStateToProps(state, ownProps) {
  return {
    decks: deckSelectors.getAllDecks(state)
  };
}

export default connect(mapStateToProps, null)(Inbox);
