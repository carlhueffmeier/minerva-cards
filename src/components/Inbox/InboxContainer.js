import { connect } from 'react-redux';
import { selectors as deckSelectors } from 'state/modules/decks';
import Inbox from './Inbox';

function mapStateToProps(state, ownProps) {
  return {
    decks: deckSelectors.getAllDecks(state)
  };
}

export default connect(mapStateToProps, null)(Inbox);
