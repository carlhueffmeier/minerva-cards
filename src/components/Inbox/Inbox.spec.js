import React from 'react';
import { shallow } from 'enzyme';
import Inbox from './Inbox';

describe('Inbox', () => {
  const props = {
    decks: [{ id: 1, title: `__TEST__` }]
  };
  const inbox = shallow(<Inbox {...props} />);

  it('renders properly', () => {
    expect(inbox).toMatchSnapshot();
  });
});
