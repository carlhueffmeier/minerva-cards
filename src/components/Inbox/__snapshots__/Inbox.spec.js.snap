// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inbox renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Inbox
    decks={
        Array [
            Object {
              "id": 1,
              "title": "__TEST__",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Inbox
</h1>,
        <ul
          className="deck-list"
>
          <li>
                    <Link
                              replace={false}
                              to="/decks/1"
                    >
                              __TEST__
                    </Link>
          </li>
</ul>,
        <Connect(NewDeck) />,
        <Button
          onClick={[Function]}
>
          Start Learning
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Inbox",
        },
        "ref": null,
        "rendered": "Inbox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                            replace={false}
                            to="/decks/1"
              >
                            __TEST__
              </Link>
</li>,
          ],
          "className": "deck-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/decks/1"
>
                __TEST__
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "__TEST__",
                "replace": false,
                "to": "/decks/1",
              },
              "ref": null,
              "rendered": "__TEST__",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Start Learning",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Start Learning",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Inbox
</h1>,
          <ul
            className="deck-list"
>
            <li>
                        <Link
                                    replace={false}
                                    to="/decks/1"
                        >
                                    __TEST__
                        </Link>
            </li>
</ul>,
          <Connect(NewDeck) />,
          <Button
            onClick={[Function]}
>
            Start Learning
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Inbox",
          },
          "ref": null,
          "rendered": "Inbox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                                replace={false}
                                to="/decks/1"
                >
                                __TEST__
                </Link>
</li>,
            ],
            "className": "deck-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/decks/1"
>
                  __TEST__
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "__TEST__",
                  "replace": false,
                  "to": "/decks/1",
                },
                "ref": null,
                "rendered": "__TEST__",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Start Learning",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Start Learning",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
